Philips.CodeAnalysis.Test/Maintainability/Maintainability/MergeIfStatementsAnalyzerTest.cs

```diff
@@ -0,0 +1,106 @@
+﻿// © 2023 Koninklijke Philips N.V. See License.md in the project root for license information.
+
+using System;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+using Philips.CodeAnalysis.Common;
+using Philips.CodeAnalysis.MaintainabilityAnalyzers.Maintainability;
+
+namespace Philips.CodeAnalysis.Test.Maintainability.Maintainability
+{
+	[TestClass]
+	public class MergeIfStatementsAnalyzerGeneratedCodeTest : DiagnosticVerifier
+	{
+		protected override DiagnosticAnalyzer GetCSharpDiagnosticAnalyzer()
+		{
+			return new MergeIfStatementsAnalyzer(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
+		}
+
+		[TestMethod]
+		public void DoNotMergeIfsGeneratedCodeTest()
+		{
+			const string testCode = @"
+		        public class MyClass
+				{
+					public void Foo()
+					{
+						if (1 == 1) 
+						{
+							if (2 == 2)
+							{ }
+						}
+					}
+			    }";
+
+			VerifyCSharpDiagnostic(testCode, "Test.Designer");
+		}
+	}
+
+
+
+	[TestClass]
+	public class MergeIfStatementsAnalyzerTest : CodeFixVerifier
+	{
+		protected override CodeFixProvider GetCSharpCodeFixProvider()
+		{
+			return new MergeIfStatementsCodeFixProvider();
+		}
+
+		protected override DiagnosticAnalyzer GetCSharpDiagnosticAnalyzer()
+		{
+			return new MergeIfStatementsAnalyzer();
+		}
+
+
+		[DataTestMethod]
+		[DataRow(@"if (1==1) { if(1==1) {} else {} }")]			// Has else clause
+		[DataRow(@"if (1==1) { int x; if (1==1) {} }")]         // Has multiple statements
+		[DataRow(@"if (1==1) { if (1==1) {} ; int x}")]         // Has multiple statements
+		[DataRow(@"{ if (1==1) {} }")]							// Parent is not if statement
```

Comment:
Consider assigning the comment text to `DisplayName`

Reply:
done, and created Issue for new analyzer idea